### PR TITLE
Sanitise users html inputs before rendering

### PR DESCRIPTION
**Pull request by piotrekpasciak**
Monday: 
https://aicrowd.monday.com/boards/363093650/pulses/458027248

Read up:
https://makandracards.com/makandra/2579-everything-you-know-about-html_safe-is-wrong

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -39,7 +39,7 @@ def footer_class
   def themed_button(opts = {})
     raise ArgumentError if opts.key?(:link) && opts.key?(:modal)
 
-    title = opts[:title].presence || '&nbsp;'.html_safe
+    title = opts[:title].presence
```
Comment (nemaniarjun) [inline app/helpers/application_helper.rb]: there should be fallback to '&nbsp;' ?

File: app/helpers/devise_helper.rb [hunk 2]
```diff
@@ -15,17 +15,16 @@ def devise_error_messages!
...
-      #{messages}
-      <button type="button" class="close" data-dismiss="alert" aria-label="Close">
-        <span aria-hidden="true">&times;</span>
-      </button>
-    </div>
-  </div>
-    HTML
 
-    html.html_safe
+    html =  "<div class='alert alert-warning alert-dismissible fade show' role='alert'>"
```
Comment (nemaniarjun) [inline app/helpers/devise_helper.rb]: no html_safe for the first line?

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (piotrekpasciak) [reply app/helpers/application_helper.rb]: I guess I can change `presence` method to `to_s`  to achieve it.

File: app/helpers/devise_helper.rb (hunk 2 above)
Comment (piotrekpasciak) [reply app/helpers/devise_helper.rb]: This method looks different now.

File: app/views/dataset_files/_toc_form.html.erb [hunk 3]
```diff
@@ -1,27 +0,0 @@
-<br/><br/><br/>
```
Comment (skbly7) [inline app/views/dataset_files/_toc_form.html.erb]: I don't find refactoring in every PR attractive approach. 

I understand it is tempting to get rid of files that aren't in use/dead code, but that needs to be separate PRs. This causes reviewing the actual/important changes meant via current PR harder.

File: spec/helpers/sanitization_helper_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (skbly7) [inline spec/helpers/sanitization_helper_spec.rb]: Nice! 💯

File: app/views/dataset_files/_toc_form.html.erb (hunk 3 above)
Comment (piotrekpasciak) [reply app/views/dataset_files/_toc_form.html.erb]: I've moved removals to separate PR, but this one is blocked until we merge the new one:
https://github.com/AIcrowd/AIcrowd/pull/1169

**Comment by piotrekpasciak** [comment]
@skbly7 Code removal is no longer part of this PR. Please take a look once again.
